cpp/src/arrow/util/thread_pool.h

```diff
@@ -276,25 +277,100 @@ class ARROW_EXPORT SerialExecutor : public Executor {
     return FutureToSync(fut);
   }
 
+  template <typename T>
+  static Iterator<T> RunGeneratorInSerialExecutor(
```

Comment:
Nit, but `SerialExecutor::RunGeneratorInSerialExecutor` seems bit verbose. Perhaps something like `SerialExecutor::IterateGenerator`?

Reply:
I like that.  I've updated the name.